app/controllers/health_cards_controller.rb

```diff
@@ -43,4 +47,22 @@ def health_card
   def find_patient
     @patient = Patient.find(params[:patient_id])
   end
+
+  def create_immunizations(bundle)
+    immunizations = bundle.entry.select { |e| e.resource.is_a?(FHIR::Immunization) }
+    vaccine_id_map = {
+      '207': 1,
+      '208': 2,
+      '212': 3
+    }
+
+    immunizations.each do |i|
```

Comment:
Will this still work completely fine if there is no immunization resource?

Reply:
It should just be an empty array if there are no immunization resources so it should be fine. The code inside the `each` just won't be executed.